src/graph/resolvers/story.js

```diff
@@ -216,14 +216,20 @@ module.exports = {
     advertiserStories: async (root, { input, pagination, sort }) => {
       const {
         advertiserId,
+        externalId,
         publisherId,
         excludeStoryIds,
       } = input;
+      // set id(advertiserId) based externalId look up or by it bein passed.
+      const id = (externalId && !advertiserId)
```

Comment:
This should throw a user input error in two cases:

If both the advertiser and the external IDs are provided 

If neither of the IDs are provided. 

There are some examples of this in some of the base CMS graphql resolvers.